src/crush/CrushWrapper.cc

```diff
@@ -226,6 +226,52 @@ int CrushWrapper::split_id_class(int i, int *idout, int *classout) const
   return 0;
 }
 
+int CrushWrapper::rename_class(CephContext *cct,
+                               const string& srcname,
+                               const string& dstname,
+                               ostream *ss) {
```

Comment:
Hmm, it kind of seems like the only use for this is when you *are* using the class.  (If you aren't, you can just remove it and create the new class.)  And the hard part about renaming a class that you are using is 

1. you'd like to preserve the bucket ids in the shadow tree across the rename so data doesn't move around, and
2. you'd like to atomically update the device classes, the class name, shadow hiearrchy, *and* the rule referencing the class.

If we aren't doing 1+2 (we aren't) then this is no better than a for loop over rm-device-class + set-device-class, right?

(Doing 1+2 would be nice, but if we don't do them, I wonder if we should just leave the rename command disabled entirely?  It won't do the hard parts the user wants it to.)